.github/workflows/prchecks.yml

```diff
@@ -70,6 +70,50 @@ jobs:
           dir.create("build")
           file.copy(binary, "build")
         shell: Rscript {0}
-    
+
       - name: Calculate code coverage
         run: Rscript -e "covr::codecov()"
+        
+  render:
+      name: Update Thesis Example and README files
+      needs: build
+      runs-on: windows-latest
+      steps:
+          # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+        - name: Checkout repository
+          uses: actions/checkout@v2
+        
+        - name: Install R
+          uses: r-lib/actions/setup-r@v1
+        
+        - name: Setup tinytex
+          uses: r-lib/actions/setup-tinytex@master
+  
+        - name: Setup pandoc
+          uses: r-lib/actions/setup-pandoc@v1
+          with:
+           pandoc-version: '2.7.3' # The pandoc version to download (if necessary) and use.
+  
+        - name: install required packages
+          run: Rscript -e 'install.packages(c("rmarkdown","remotes"))'
+          
+        - name: install iheiddown package
+          run: Rscript -e 'remotes::install_github("jhollway/iheiddown")'
+  
+        - name: render README
+          run: Rscript -e 'rmarkdown::render("README.Rmd", output_format = "md_document")'
+          
+        - name: commit rendered README
+          run: |
+            git add README.md
+            git commit -m "Re-build README.md" || echo "No changes to commit"
+            git push origin master || echo "No changes to commit"
+          
+        - name: Render output samples
+          run: Rscript R/_render.R
+          
+        - name: commit rendered Thesis Example
+          run: |
+            git add inst/rmarkdown/templates/thesis/skeleton/versions/IHEIDThesisExample.pdf
+            git commit -m "Re-build Thesis Example" || echo "No changes to commit"
+            git push origin master || echo "No changes to commit"
```

Comment:
But is this necessary on checks, or only on release? We want to avoid a bunch of random GitHub commits that just duplicate info.

Reply:
It isn't, I added it just to see if the chunk was running. The idea is to knit all examples only on changes to the main branch.